cypress/e2e/GUI/sorts_by_number.cy.js

```diff
@@ -0,0 +1,35 @@
+describe('Teste de ordenação por número de funcionários', () => {
```

Comment:
É impossível entender o que estes testes fazem sem perder um certo tempo lendo-os e entendendo o que acontece entre as linhas 7-15.

**Testes devem ser fáceis de ler e devem ser** (em 99,99% dos casos) **lineares.**

Ou seja, sem `for` _loops_, condicionais, etc.

Veja abaixo uma alternativa (de testes de componentes), onde passo como _prop_ dados conhecidos à tabela, podendo saber exatamente quais dados devem estar na primeira e última linha, por exemplo, para garantir o funcionamento da ordenação.

```js
import Table from './Table'

const customers = require('../../../cypress/fixtures/customers.json').customers

describe('<Table />', () => {
  beforeEach(() => {
    cy.mount(<Table customers={customers} customerClickHandler={cy.stub()} />)
  })

  it('sorts by Size in descending order by default', () => {
    cy.contains('th', 'Size ↓').should('be.visible')
    cy.get('tbody tr')
      .first()
      .find('td')
      .eq(3)
      .should('contain', 'Very Large Enterprise')

    cy.get('tbody tr')
      .last()
      .find('td')
      .eq(3)
      .should('contain', 'Small')
  })

  it('sorts by Size in ascending order', () => {
    cy.contains('th', 'Size').click()
    cy.contains('th', 'Size ↑').should('be.visible')
    cy.get('tbody tr')
      .first()
      .find('td')
      .eq(3)
      .should('contain', 'Small')

    cy.get('tbody tr')
      .last()
      .find('td')
      .eq(3)
      .should('contain', 'Very Large Enterprise')
  })

  it('sorts by Number of employees in descending order', () => {
    cy.contains('th', 'Number of employees').click()
    cy.contains('th', 'Number of employees ↓').should('be.visible')
    cy.get('tbody tr')
      .first()
      .find('td')
      .eq(2)
      .should('contain', '51015')

    cy.get('tbody tr')
      .last()
      .find('td')
      .eq(2)
      .should('contain', '47')
  })

  it('sorts by Number of employees in ascending order', () => {
    cy.contains('th', 'Number of employees').click()
    cy.contains('th', 'Number of employees').click()
    cy.contains('th', 'Number of employees ↑').should('be.visible')
    cy.get('tbody tr')
      .first()
      .find('td')
      .eq(2)
      .should('contain', '47')

    cy.get('tbody tr')
      .last()
      .find('td')
      .eq(2)
      .should('contain', '51015')
  })
})

```

Recomendo reconsiderar sua implementação, e caso queri continuar com testes E2E, ao menos use o comando `cy.intercept()` devolvendo dados conhecidos ao frontend (com o uso de _fixtures_), para testar de forma linear, como um usuário, não como um programador. 😛